packages/components/ellipsize-text/src/ellipsize-text.ts

```diff
@@ -0,0 +1,78 @@
+import { type ScopedElementsMap, ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';
+import { Tooltip } from '@sl-design-system/tooltip';
+import { type CSSResultGroup, LitElement, type TemplateResult, html } from 'lit';
+import styles from './ellipsize-text.scss.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-ellipsize-text': EllipsizeText;
+  }
+}
+
+/**
+ * Small utility component to add ellipsis to text that overflows
+ * its container. It also adds a tooltip with the full text.
+ */
+export class EllipsizeText extends ScopedElementsMixin(LitElement) {
+  /** @internal */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-tooltip': Tooltip
+    };
+  }
+
+  /** @internal */
+  static override styles: CSSResultGroup = styles;
+
+  /** Observe size changes. */
+  #observer = new ResizeObserver(() => this.#onResize());
+
+  /** The lazy tooltip. */
+  #tooltip?: Tooltip | (() => void);
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.#observer.observe(this);
+  }
+
+  override disconnectedCallback(): void {
+    this.#observer.disconnect();
+
+    if (this.#tooltip instanceof Tooltip) {
+      this.#tooltip.remove();
+    } else if (this.#tooltip) {
+      this.#tooltip();
+    }
+
+    this.#tooltip = undefined;
+
+    super.disconnectedCallback();
+  }
+
+  override render(): TemplateResult {
+    return html`<slot></slot>`;
+  }
+
+  #onResize(): void {
+    if (this.offsetWidth < this.scrollWidth) {
+      this.#tooltip ||= Tooltip.lazy(
```

Comment:
Maybe the tooltip should be also available on focus, not only on hover?

Reply:
So the whole element should be focusable, when there is ellipsis?

Reply:
The tooltip also listens for `focusin`, so that is already covered :) I think the focus part also needs to wait until we have #1526 so we can test it properly.